### PR TITLE
HPCC-29838 Add false sharing test case

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by mckellyln** [comment]
Do you get results similar to this ?
20:52:51.083809 2699800 Same RAs took 6105ms
20:52:57.159401 2699800 Adjacent RAs took 6078ms
20:53:01.858301 2699800 Spaced RAs took 4701ms
I don't have a dual-socket machine to see if any addl CC effect.

**Review by mckellyln — APPROVED** [review]
Approved.
Perhaps also interesting if near and far non-atomic increment were added as well.

**Comment by GordonSmith** [comment]
@richardkchapman let me know if your happy for it be merged (I will take Gavins comment as an approval!).

File: roxie/ccd/ccdsnmp.cpp [hunk 1]
```diff
@@ -37,10 +37,10 @@ RoxieQueryStats loQueryStats;
 RoxieQueryStats hiQueryStats;
 RoxieQueryStats slaQueryStats;
 RoxieQueryStats combinedQueryStats;
-RelaxedAtomic<unsigned> retriesIgnoredPrm;
-RelaxedAtomic<unsigned> retriesIgnoredSec;
-RelaxedAtomic<unsigned> retriesNeeded;
-RelaxedAtomic<unsigned> retriesReceivedPrm;
+RelaxedAtomic<unsigned> retriesIgnoredPrm __attribute__((aligned(CACHE_LINE_SIZE)));
```
Comment (richardkchapman) [inline roxie/ccd/ccdsnmp.cpp:40]: This change should probably not be included

File: system/jlib/jatomic.hpp [hunk 2]
```diff
@@ -51,7 +51,7 @@ auto sub_fetch(T & value, decltype(value.load()) delta, std::memory_order order
 //NOTE: Counts will never be lost, but the values read from another thread may be inconsistent.
 //E.g., thread 1 updates x than y, thread 2 may read an updated value of y, but an old value of x.
 template <typename T>
-class RelaxedAtomic : public std::atomic<T>
+class alignas(CACHE_LINE_SIZE) RelaxedAtomic : public std::atomic<T>
```
Comment (richardkchapman) [inline system/jlib/jatomic.hpp:54]: Nor this one

**Comment by richardkchapman** [comment]
I removed a couple of changes that should not really have been included. As far as I am concerned it's now ok to merge, but not in any way urgent.

**Comment by GordonSmith** [comment]
@mckellyln can you re-review latest changes?

**Review by mckellyln — APPROVED** [review]
I am fine with this PR except I think the java changes are not related and perhaps would go away with a rebase ?

**Comment by richardkchapman** [comment]
@mckellyln I have repushed to remove the unwanted Java changes

**Comment by mckellyln** [comment]
@GordonSmith good to merge now

**Pull request merged**